docs/modules/ROOT/pages/udc.adoc

```diff
@@ -1,226 +1,144 @@
+Instead, only the class hash, salt, and constructor arguments determine the address.
+This type of deployment enables redeployments of accounts and known systems across multiple networks.
+To deploy a reproducible deployment, set `from_zero` to `true`.
 
-[,cairo]
+[source,cairo]
 ----
-classHash: felt
-salt: felt
-unique: felt
-calldata_len: felt
-calldata: felt*
+let deployed_addr = udc.deploy_contract(class_hash, salt, true, calldata.span());
 ----
 
-Returns:
+== Version changes
 
-[,cairo]
-----
-address: felt
-----
+:pedersen: link:https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#pedersen_hash[Pedersen]
+:poseidon: link:https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#poseidon_hash[Poseidon]
+:origin-dependent-deployments: xref:origin_dependent[origin-dependent deployments]
+:previous-udc-api: link:https://docs.openzeppelin.com/contracts-cairo/0.6.1/udc#api_specification[previous Universal Deployer API]
 
-=== Events
+NOTE: See the {previous-udc-api} for the initial spec.
 
-[,cairo]
-----
-func ContractDeployed(
-    address: felt,
-    deployer: felt,
-    unique: felt,
-    classHash: felt,
-    calldata_len: felt,
-    calldata: felt*,
-    salt: felt
-) {
-}
-----
+The latest iteration of the UDC includes some notable changes to the API which include:
 
-==== `ContractDeployed`
+* `deployContract` method is replaced with the snake_case {deploy_contract}.
+* {pedersen} hashing algorithm is replaced with the {poseidon} hashing algorithm for computing the salt on {origin-dependent-deployments}.
+* `unique` parameter is replaced with `from_zero` in both the `deploy_contract` method and `ContractDeployed` event.
 
-Emitted when `deployer` deploys a contract through the Universal Deployer Contract.
+WARNING: Origin-dependent deployments previously meant that the `unique` argument would be `true`.
+Origin-dependent deployments from the new UDC iteration, however, requires that `from_zero` is `false`.
+It's especially important to keep this in mind when dealing with `ContractDeployed` events because this change will appear as the opposite boolean per deployment type.
 
-Parameters:
+== Precomputing contract addresses
 
-[,cairo]
-----
-address: felt,
-deployer: felt,
-unique: felt,
-classHash: felt,
-calldata_len: felt,
-calldata: felt*,
-salt: felt
-----
+:calculate_address_from_zero: xref:/utilities.adoc#universal_deployer-calculate_contract_address_from_zero[calculate_address_from_zero]
+:calculate_address_not_from_zero: xref:/utilities.adoc#universal_deployer-calculate_contract_address_not_from_zero[calculate_address_not_from_zero]
+
+This library offers utility functions written in Cairo to precompute contract addresses.
+They include {calculate_address_from_zero} and {calculate_address_not_from_zero} to return origin-independent and origin-dependent contract addresses respectively.
```

Comment:
As I suggested for the util, `calculate_address_not_from_zero` should have a name more specific to UDC, because this could be also used for the `deploy_syscall`, being the name kind of generic, and the current implementation is UDC specific. I think we should add a helper for the UDC and link it here while keeping a generic one.